packages/task/src/node/task-server.ts

```diff
@@ -0,0 +1,274 @@
+                        reject(new Error(error));
+                    }
+                })
+                .catch(err => {
+                    reject(new Error(err));
+                });
+
+        });
+    }
+
+    protected fireTaskExitedEvent(event: ITaskExitedEvent) {
+        this.logger.debug(log => log(`task has exited:`, event));
+        // notify all clients. They can filter on context to see if it's relevant to them
+        this.clients.forEach(client => {
+            client.onTaskExit(event);
+        });
+    }
+
+    protected fireTaskCreatedEvent(event: ITaskInfo) {
+        this.logger.debug(log => log(`task created:`, event));
+        // notify all clients. They can filter on context to see if it's relevant to them
+        this.clients.forEach(client => {
+            client.onTaskCreated(event);
+        });
+    }
+
+    kill(id: number): Promise<void> {
+        const taskToKill = this.taskManager.get(id);
+        if (taskToKill !== undefined) {
+            taskToKill.kill();
+            this.logger.info(`Attempting to kill task id ${id}`);
+        }
+        return Promise.resolve();
+    }
+
+    /** Set the client we send the notifications-to. We expect that there might be
+     * multiple clients, so add them to a list as they are set
+     */
+    setClient(client: ITaskClient) {
+        this.clients.push(client);
+    }
+
+    /**
+     * uses heuristics to look-for a command. Will look into the system path, if the command
+     * is given without a path. Will resolve if a potential match is found, else reject. There
+     * is no garantee that a command we find will be the one executed, if multiple commands with
+     * the same name exist.
+     * @param command command name to look-for
+     * @param cwd current working directory
+     */
+    private async findCommand(command: string, cwd: string): Promise<string> {
+        const systemPath = process.env.PATH;
+
+        return new Promise<string>(async (resolve, reject) => {
+
+            if (path.isAbsolute(command)) {
+                if (await this.fileExists(command)) {
+                    resolve(command);
+                    return;
+                } else {
+                    const uri = new URI(cwd).resolve(command);
+                    const resolved_command = FileUri.fsPath(uri);
+                    if (await this.fileExists(resolved_command)) {
+                        resolve(resolved_command);
+                        return;
+                    }
+                }
+            } else {
+                // look for command relative to cwd
+                const resolved_command = FileUri.fsPath(new URI(cwd).resolve(command));
+
+                if (await this.fileExists(resolved_command)) {
+                    resolve(resolved_command);
+                    return;
+                } else {
+                    // should cover Unix and Windows cases
+                    const separator = /;|:/;
```

Comment:
`:` is valid as a linux file we should check for isWindows and select 1 separator only.

You can use: https://nodejs.org/api/path.html#path_path_delimiter

```
➜  ~ touch ab:cd
➜  ~ ls -la ./ab:cd 
-rw-r--r-- 1 eantotr eusers 0 Dec  4 14:22 ./ab:cd
➜  ~ touch abc;cd
➜  ~ ls -l ab*
-rw-r--r-- 1 eantotr eusers 0 Dec  4 14:23 abc
-rw-r--r-- 1 eantotr eusers 0 Dec  4 14:22 ab:cd

```

Reply:
Cool, I didn't know about path.delimiter. Since we get that value from node, I think it already takes into account which platform we are on. 

Done (except handling the case where there is a ":" used within a directory in the system path - Sorry, if you do that, you deserve for your executable not to be found ;)

Reply:
>except handling the case where there is a ":" used within a directory in the system path - Sorry, if you do that, you deserve for your executable not to be found

That won't happen since I guess on windows you can't create a file or directory with `:` like you can't do a file /dir with `;` in linux ?